packages/code/src/Code.tsx

```diff
@@ -169,16 +280,55 @@ function Code({
     </Syntax>
   );
 
+  function handleScroll(e: React.UIEvent) {
+    const scrollWidth = e.currentTarget.scrollWidth;
+    const elementWidth = e.currentTarget.clientWidth;
+    const isScrollable = scrollWidth > elementWidth;
+
+    if (isScrollable) {
+      const scrollPosition = e.currentTarget.scrollLeft;
+      const maxPosition = scrollWidth - elementWidth;
+      setCanScrollLeft(scrollPosition > 0);
```

Comment:
Would it be easier/more readable to combine and do something like [scroll, setScroll] = useState('none') and have it be an enum of values: `{'left', 'right', 'both', 'none'}`. You don't have to run with this, just a suggestion that may help clean things up a bit

Reply:
This ended up becoming much cleaner once I started using an enum 👍